libethcore/Precompiled.cpp

```diff
@@ -99,48 +99,77 @@ ETH_REGISTER_PRECOMPILED(identity)(bytesConstRef _in)
 
 // Parse _count bytes of _in starting with _begin offset as big endian int.
 // If there's not enough bytes in _in, consider it infinitely right-padded with zeroes.
-bigint parseBigEndianRightPadded(bytesConstRef _in, size_t _begin, size_t _count)
+bigint parseBigEndianRightPadded(bytesConstRef _in, bigint _begin, bigint _count)
 {
-	if (_begin > _in.count())
+	if (_begin > numeric_limits<size_t>::max())
```

Comment:
No, I think you should compare `if (_begin > _in.count())` instead of `if (begin > _in.count())` and get rid of at least the first test.

In fact having the huge `_begin` should not be a problem for this function, since we know that input is infinitely right-padded with zeroes, it should return 0. It's not like this is an invalid `_begin` argument